### PR TITLE
Update 1.13.1

### PR DESCRIPTION
**Pull request by danpetry**
# Links

[Upstream](https://github.com/pytorch/pytorch)
[Upstream update diff](https://github.com/pytorch/pytorch/compare/v1.12.1...v1.13.1)
[Conda-forge update diff](https://github.com/conda-forge/pytorch-cpu-feedstock/compare/93d901f13b3ac87f6c63d143d53ec0000977c729...24ff4b36cee2dcdffc095b2cee6d6d5a414be70d)
[Changelog](https://github.com/pytorch/pytorch/releases)
[Jira ticket](https://anaconda.atlassian.net/browse/PKG-1474)
[License](https://github.com/pytorch/pytorch/blob/master/LICENSE)

# Changes
- Bump version, update checksum
- add —no-build-isolation to ignore the upstream pyproject.toml
- remove blas stuff in cbc.yaml, because it’s now in the aggregate cbc.yaml
- rebase patches and remove upstreamed ones
- don’t build for python 3.11 on PPC. PPC needs to be built with gcc v8, to avoid linker errors (see cbc.yaml in this feedstock). This, however, results in run dependency conflicts with python 3.11, which has been built with gcc v11. The conflicts concern ld_impl_linux-ppc64le, which is put into the runtime environment indirectly by gcc. See git history for numerous attempts to find the right flags to solve this issue with gcc v11.
- include numpy pinning for python 3.11
- include host pinnings

# Notes
- Currently neither Prefect nor Concourse can build this feedstock for Windows. Team Sirius and IT, respectively, are working on solutions. I propose delivering all other platforms and then delivering for Windows when this is solved.

### CODE REVIEW

File: recipe/meta.yaml
Comment (ELundby45) [inline recipe/meta.yaml:9]: Do we want to be using the tarball including submodules for release?

**Review by ELundby45 — APPROVED** [review]
Overall LGTM, just one curiosity question. I would appreciate a review from someone more familiar with this feedstock.

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml:45]: Are there missing `entry_points`? See https://github.com/pytorch/pytorch/blob/49444c3e546bf240bed24a101e747422d1f8a0ee/setup.py#L944-L950

**Comment by skupr-anaconda** [comment]
@dholth Seems that the error on win-64  caused by `conda_package_handling`:
`conda_package_handling.exceptions.ArchiveCreationError: info/test/test/functorch: Cannot stat while writing file: info\test\test\functorch`
@danpetry Or do we have issues with CI specifically?

**Comment by dholth** [comment]
I think we may have introduced a bug in the conda-build release (from about 16 days ago) that keeps a file open. Then we see the same problem in conda-package-handling < 2 and > 2.

**Review by varlackc — COMMENTED** [review]
The package is currently failing for `win-64` we will have to investigate the issue further before we can approve it.

File: recipe/meta.yaml
Comment (danpetry) [reply recipe/meta.yaml:9]: good question. Probably? If we use the version that includes the submodules, there's less chance that one of the submodule projects could re-release a version over the top and break something, or remove the project from where it was, etc

**Comment by danpetry** [comment]
@dholth thanks for the investigation on this
@varlackc please see above discussion and description. PO has cleared us to go ahead without Windows for the time being, and build it out later

File: recipe/meta.yaml
Comment (danpetry) [reply recipe/meta.yaml:45]: thanks, added

File: recipe/meta.yaml
Comment (danpetry) [reply recipe/meta.yaml:116]: The issue with `mkl` on `osx-64` isn't because we don't build it. It's because it was a pain in the butt trying to fix runtime issues caused by conflicts between `openblas` and `mkl`, and nobody was really asking for pytorch `mkl`+`osx-64`.

As an aside, `mkl` is available on `osx-64`, just not the most recent version. For my own understanding, would you mind explaining why this release would unblock?
```
dpetry@Daniels-MacBook-Pro ~/Sandbox/aggregate (master) $ conda search mkl[subdir=osx-64]
Loading channels: done
# Name                       Version           Build  Channel             
mkl                         2017.0.4      h1fae6ae_0  pkgs/main           
mkl                         2018.0.0      h5ef208c_6  pkgs/main           
mkl                         2018.0.0      hc285769_4  pkgs/main           
mkl                         2018.0.1      hfbd8650_4  pkgs/main           
mkl                         2018.0.2               1  pkgs/main           
mkl                         2018.0.3               1  pkgs/main           
mkl                           2019.0             117  pkgs/main           
mkl                           2019.0             118  pkgs/main           
mkl                           2019.1             144  pkgs/main           
mkl                           2019.3             199  pkgs/main           
mkl                           2019.4             233  pkgs/main           
mkl                           2020.0             166  pkgs/main           
mkl                           2020.1             216  pkgs/main           
mkl                           2020.2             260  pkgs/main           
mkl                         2021.2.0    hecd8cb5_269  pkgs/main           
mkl                         2021.3.0    hecd8cb5_517  pkgs/main           
mkl                         2021.4.0    hecd8cb5_637  pkgs/main           
mkl                         2022.0.0    hecd8cb5_105  pkgs/main           
mkl                         2022.1.0    hf3af037_208  pkgs/main           
mkl                         2022.1.0    hf3af037_209  pkgs/main
```

File: recipe/meta.yaml
Comment (danpetry) [reply recipe/meta.yaml:45]: Would added entry points cause a build to fail? I'm just wondering if I need to run the CI on this PR again before merge, after this change.

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:116]: Ah, thanks!

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Pull request merged**
